src/Microsoft.Health.Fhir.Core/Features/Operations/ConvertData/Models/InputDataTypeAndDefaultTemplateCollectionInconsistentException.cs

```diff
@@ -0,0 +1,25 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Diagnostics;
+using Microsoft.Health.Fhir.Core.Exceptions;
+using Microsoft.Health.Fhir.Core.Models;
+
+namespace Microsoft.Health.Fhir.Core.Features.Operations.ConvertData.Models
+{
+    public class InputDataTypeAndDefaultTemplateCollectionInconsistentException : FhirException
```

Comment:
>InputDataTypeAndDefaultTemplateCollectionInconsistentException [](start = 17, length = 62)

Is there are any benefit of creating this exception comparing to just use `RequestNotValidException`?

Reply:
The response message of RequestNotValidException is "Value of the following parameter {0} is invalid.". There is only one parameter in the message, while there are two parameters in the inconsistent message.

Reply:
Then I look on this exception and `RequestNotValidException` only difference I see is name of the class, they both accept strings, I understand what you want to display different string for case of inconsistency, but you can just pass different string to already established exception instead of creating copy of it.

---
In reply to: [597468434](https://github.com/microsoft/fhir-server/pull/1756#discussion_r597468434) [](ancestors = 597468434)

Reply:
I removed `InputDataTypeAndDefaultTemplateCollectionInconsistentException`, and use `RequestNotValidException` now.